apps/ideaspace/src/components/modules/DashboardPage/DashboardPage.js

```diff
@@ -25,21 +24,28 @@ function DashboardPage() {
 
   React.useEffect(async () => {
     if (isAuthenticated) {
-      const data = cleanDataList(await agent.Ideas.get(
-        new URLSearchParams(`populate=*pagination[pageSize]=1000`)));
+      const data = cleanDataList(
+        await agent.Ideas.get(new URLSearchParams(`populate=deep`))
+      );
+
+      const allCards = data.map((item) => {
+        if (item.comments === undefined) item.comments = [];
+        else item.comments = cleanDataList(item.comments.data);
 
-        const allCards = data.map((item) => {
-          item.comments = cleanDataList(item.comments.data);
-          return {
-            ...item,
-            mostRecentCommentTime: new Date(
-              item.comments[0]?.updated_at
-            ).getTime(),
-          };
-        });
+        if (item.author.data !== null) {
+          item.author = cleanData(item.author.data);
+        }
 
-        setLoading(false);
-        setSourceCards(allCards);
+        return {
+          ...item,
+          mostRecentCommentTime: new Date(
+            item.comments[0]?.updated_at
```

Comment:
Do we need to search through all comments?

Reply:
> Do we need to search through all comments?

@chungthuang it's a requirement to count the total number of comments for a given user.

Reply:
Sorry my question was if we have a list of comments, don't we need to iterate through the list to find the most recent comment time? The current logic just check the first comment. Or is the API response already sorted by updated  time?

Reply:
Oh, I see. We can test to make sure the API response is sorted by updated time. Otherwise I think you are correct and we need to make an update.